app/services/login.js

```diff
@@ -0,0 +1,31 @@
+import Service from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { AUTH } from '../constants/urls';
+
+export default class LoginService extends Service {
+  @service store;
+  @tracked isLoggedIn;
+  @tracked userData;
+  @tracked isLoading = true;
+
+  constructor() {
+    super(...arguments);
+    this.checkAuth();
+  }
+
+  async checkAuth() {
+    try {
+      const user = await this.store.findRecord('user', 'self');
+      if (user) {
+        if (user.incompleteUserDetails) window.location.replace(AUTH.SIGN_UP);
+        this.isLoggedIn = true;
+        this.userData = user;
+      }
+    } catch (error) {
+      this.isLoggedIn = false;
+    } finally {
+      this.isLoading = false;
+    }
```

Comment:
Since the finally block will execute post try and catch block.

1. it will overwrite the isLoggedIn from true to false which we set in the try block. Is that the desired behavior?
2. In catch we can use console the error using `console.error`and we need not set `isLoggedIn` false as it will be set false in finally block.

Reply:
`this.isLoggedIn` and `this.isLoading` are two different tracked properties.
- `this.isLoading` is set to false in `finally` block
- `this.isLoggedIn` is set to true in `try` and false in `catch` block

Reply:
For the 2nd comment, I have added the `console.error`

Reply:
My Bad I confused `isLoading` and `isLoggedIn`. Right, in that case, it's fine what you have done.